### PR TITLE
[WIP] Xception Backend 

### PR DESCRIPTION
**Pull request by roya0045**
Draft PR for a Xception backend proposal

### CODE REVIEW

File: mmdet/models/backbones/xception.py [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from collections import OrderedDict
+from mmcv.cnn import build_conv_layer, build_norm_layer
+
+from ..builder import BACKBONES
+
+
+class SeparableConv(nn.Module):
```
Comment (ZwwWayne) [inline mmdet/models/backbones/xception.py:36]: May add registry here and make it an instance of `CONV_LAYERS`.

File: mmdet/models/backbones/xception.py [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+        x += skip
+        return x
+
+
+@BACKBONES.register_module()
+class Xception(nn.Module):
+    """Xception optimized for the ImageNet dataset, as specified in
+    https://arxiv.org/pdf/1610.02357.pdf."""
+
+    def __init__(self,
```
Comment (ZwwWayne) [inline mmdet/models/backbones/xception.py:172]: 1. Please add complete docstrings for all newly added modules and functions as those in MMDetection V2.x.

File: mmdet/models/backbones/xception.py [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+            self.conv1.out_channels,
+            self.conv_sizes[1],
+            3,
+            bias=False)
+        if self.bn:
+            self.bn2_name, self.bn2 = build_norm_layer(self.norm_cfg,
+                                                       self.conv2.out_channels)
+        # do relu here
+
+        self.block1 = Block(
```
Comment (ZwwWayne) [inline mmdet/models/backbones/xception.py:206]: 1. Use `self.layer1`, `self.layer2`, which are more consistent with ResNets.
2. Can initialize the blocks/stage layers in a more general form as [that in ResNet](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/backbones/resnet.py#L421)?

File: mmdet/models/backbones/xception.py [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+                                   self.conv_sizes[7], 3, 1, 1)
+        if self.bn:
+            self.bn4_name, self.bn4 = build_norm_layer(self.norm_cfg,
+                                                       self.conv4.out_channels)
+
+        # self.fc = nn.Linear(2048, num_classes)
+
+    def _init_weights(self, ):
+        for module in self.modules():
+            if isinstance(module, nn.Conv2d):
```
Comment (ZwwWayne) [inline mmdet/models/backbones/xception.py:274]: Use the implemented initialization methods (constant_init, kaiming_init) implemented in MMCV, unless the initialization is special and not implemented in MMCV. If so, please also add comments in the code as a reminder.

**Comment by ZwwWayne** [comment]
Something to do before merge:
1. For the backbone, usually it should have a corresponding PR in MMCV to add pre-trained checkpoints. These checkpoints should have been trained on ImageNet and have similar performance (accuracy) as those reported in the paper (for correctness). They can be either adopted from the officially released model (by converting keys to the current correct implementation) or trained by the community. And the implementation in MMDetection should be able to correctly load these keys with an extra classification head (not implemented by mmdet) and produce similar accuracy as the pre-trained classifier.
2. An example of this process is [RegNet](https://github.com/open-mmlab/mmdetection/tree/master/configs/regnet).

**Comment by roya0045** [comment]
@ZwwWayne Thank you for the review. I have addressed partly your comments for the moment. I will do some more once I get more time. Thanks once again!

**Comment by OpenMMLab-Assistant003** [comment]
Hi @roya0045！We are grateful for your efforts in helping improve this open-source project during your personal time. 
Welcome to join OpenMMLab Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA
If you have a WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤ @roya0045
